cmd/deploy.go

```diff
@@ -131,44 +161,85 @@ func createReleaseFile() (filename string, err error) {
 	}
 
 	var hasGit bool = true
-	if _, err = exec.LookPath("git"); err != nil {
+	if errGit := d.LookPath(d.git); errGit != nil {
 		hasGit = false
 	}
 
-	if _, err = os.Stat(".git"); hasGit && !os.IsNotExist(err) {
-		// we are in a Git environment
-		var (
-			output []byte
-			files  []string
-		)
-		// Exclude list
-		// git ls-files -d // delete files
-		output, err = exec.Command("git", "ls-files", "-d").CombinedOutput()
-		if err != nil {
-			panic(fmt.Errorf("Failed listing deleted "))
-		}
-		tarball.SetIgnoreList(strings.Split(string(output), "\n"))
+	if _, errGit := os.Stat(".git"); !hasGit || os.IsNotExist(errGit) {
+		// not a GIT repo/environment!
+		d.Println("Fallback to tarball full current working directory...")
+		cwd, _ = os.Getwd()
+		filename, err = tarball.CompressFolder(cwd)
+		return
+	}
+
+	// we are in a GIT environment!
+	var (
+		files, allFiles []string
 
-		// Include list
-		// git ls-files -c
-		output, err = exec.Command("git", "ls-files", "-c").CombinedOutput()
-		if err != nil {
-			panic(fmt.Errorf("Failed list Git cached files"))
+		gitListingFilesFlags = [][]string{
+			// Include commited files - git ls-files -c
+			{"-c"},
+
+			// Untracked files - git ls-files -o --exclude-standard
+			{"-o", "--exclude-standard"},
 		}
-		files = append(files, strings.Split(string(output), "\n")...)
-		// git ls-files -o --exclude-standard
-		output, err = exec.Command("git", "ls-files", "-o", "--exclude-standard").CombinedOutput()
-		if err != nil {
-			panic(fmt.Errorf("Failed list Git untracked non-ignored files"))
+	)
+
+	// Exclude list - git ls-files -d
+	if files, err = d.parseFilesListFromGIT([]string{"-d"}); err != nil {
+		return
+	}
+	tarball.SetIgnoreList(files)
+
+	for _, lsArgs := range gitListingFilesFlags {
+		if files, err = d.parseFilesListFromGIT(lsArgs); err != nil {
+			return
 		}
-		files = append(files, strings.Split(string(output), "\n")...)
 
-		filename, err = tarball.CompressFiles(files)
-	} else {
-		fmt.Println("Fallback to tarball full current working directory...")
-		cwd, _ = os.Getwd()
-		filename, err = tarball.CompressFolder(cwd)
+		allFiles = append(allFiles, files...)
 	}
+	filename, err = tarball.CompressFiles(d.handleDeployEnv(allFiles))
+	return
+}
 
+func (d *KoolDeploy) parseFilesListFromGIT(args []string) (files []string, err error) {
+	var output string
+
+	output, err = d.Exec(d.git, append([]string{"ls-files", "-z"}, args...)...)
+	if err != nil {
+		d.Error(err)
+		err = fmt.Errorf("failed listing GIT files")
```

Comment:
I think it should be better if we concat the error here instead of printing two messages